src/temperature/crud.py

```diff
@@ -0,0 +1,69 @@
+import asyncio
+from datetime import datetime
+
+from httpx import AsyncClient
+from sqlalchemy import select, insert
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.orm import selectinload
+
+from src.city.crud import get_all_cities
+from src.city.models import DBCity
+from src.temperature.models import DBTemperature
+from weather_api import get_api_temperature
+
+
+async def set_cities_temperature(db: AsyncSession):
+    cities = await get_all_cities(db)
+    async with AsyncClient() as client:
+        tasks = [
+            asyncio.create_task(create_or_update_db_temperature(db, city, client))
+            for city in cities
+        ]
+        await asyncio.gather(*tasks)
+
+    await db.commit()
+
+    return await list_cities_temperature(db)
+
+
+async def list_cities_temperature(db: AsyncSession, city_id: int = None):
+    query = select(DBTemperature).options(selectinload(DBTemperature.city))
+    if city_id:
+        query = query.filter(DBTemperature.city_id == city_id)
+
+    result = await db.execute(query)
+
+    return [city[0] for city in result.fetchall()]
```

Comment:
The list comprehension `[city[0] for city in result.fetchall()]` assumes that each row in the result contains only one element. This is correct if `DBTemperature` is the only column selected. Ensure this assumption holds true in all contexts where this function is used.